AnimalKingdom/src/animalKingdom/Animal.java

```diff
@@ -0,0 +1,36 @@
+package animalKingdom;
+
+public abstract class Animal {
+
+    String name;
+    int Id;
+    int year;
+
+    public Animal(String name, int Id, int year) {
+        this.name = name;
+        this.Id = Id;
+        this.year = year;
```

Comment:
Instead of having to manually supply an ID value (which does not seem to be checked for uniqueness here), you can set up a `private static int` variable (perhaps name it something like `maxID` or `nextID`), assign it to the `id` variable upon every run of the constructor(s) and then increment it afterwards. For illustration:
```
    private static int nextID = 0;

    public Animal(String name, int year) {
        this.id = maxID;
        maxID++;
        this.name = name;
        this.year = year;
    }
```